src/classes/ValidatedTopic.ts

```diff
@@ -504,6 +521,68 @@ export default class ValidatedTopic {
     return this.getState().archivist.del(this.getTopic(), this.getIndex())
   }
 
+  /**
+   * Check if the current topic is locked.
+   */
+  public async isLocked(state: mage.core.IState = new mage.core.State()) {
+    return new Promise((resolve, reject) => {
+      state.archivist.get(this.getTopic(), this.getLockIndex(), {
+        optional: true
+      }, (error, isLocked) => {
+        if (error) {
+          return reject(error)
+        }
+
+        resolve(!!isLocked)
+      })
+    })
+  }
+
+  /**
+   * Lock this topic instance
+   *
+   * By default, locked topics will be unlocked
+   * once `state.distribute` is called on the state associated
+   * with this topic. If you do not want this behaviour,
+   * set `autoUnlock` to true, and make sure
+   * to call `unlock` manually when you are done.
+   *
+   * @param autoUnlock Unlock this topic once we complete the transaction
+   */
+  public async lock(autoUnlock: boolean = true) {
```

Comment:
I interpreted autoUnlock as "unlock this lock .... perhaps after a timeout or something". But it's really about a previously existing lock, right? I don't know what the best name is, but one suggestion would be calling it `force` instead of `autoUnlock`.